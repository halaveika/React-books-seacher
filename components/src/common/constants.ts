export const POSTER_URL = '/assets/images/';
export const SEARCH_VALUE_LOCALSTORAGE = 'components_searchValue12345';
