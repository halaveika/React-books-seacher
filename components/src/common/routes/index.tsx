import React from 'react';
import AboutPage from '../../pages/about';
import MainPage from '../../pages/main';
import NotFoundPage from '../../pages/not-found';

const routes = [
  { path: '/', element: <MainPage title="cards" /> },
  { path: '/about', element: <AboutPage title="about" /> },
  {
    path: '/404',
    element: <NotFoundPage title="not found page" />,
  },
];

export default routes;
