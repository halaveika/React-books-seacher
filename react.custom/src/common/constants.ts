export const POSTER_URL = '/assets/images/';
export const SEARCH_VALUE_LOCALSTORAGE = 'components_searchValue12345';
export const FORM_CARDS = 'react.forms_cards12345';
export const GOOGLE_URL = 'https://www.googleapis.com/books/v1/';
export const PAGE_SIZE_ARR = [10, 20, 30, 40];
export const FILTER_ARR = ['all', 'books ', 'magazines '];
