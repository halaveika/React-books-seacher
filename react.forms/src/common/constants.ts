export const POSTER_URL = '/assets/images/';
export const SEARCH_VALUE_LOCALSTORAGE = 'components_searchValue12345';
export const FORM_CARDS = 'react.forms_cards12345';
