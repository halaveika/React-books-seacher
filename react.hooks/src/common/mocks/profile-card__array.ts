export const mockCardArr = [
  {
    id: 'test1-id',
    firstname: 'firstname',
    lastname: 'lastname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test1-gender',
    avatar: 'test1-avatar',
    city: 'test1-city',
    gifts: [],
  },
  {
    id: 'test2-id',
    firstname: 'test2-firstname',
    lastname: 'test2-firstname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test2-gender',
    avatar: 'test2-avatar',
    city: 'test2-city',
    gifts: ['test2-gift1', 'test2-gift2'],
  },
  {
    id: 'test3-id',
    firstname: 'test3-firstname',
    lastname: 'test3-firstname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test3-gender',
    avatar: 'test3-avatar',
    city: 'test3-city',
    gifts: ['test3-gift1', 'test3-gift2'],
  },
];
