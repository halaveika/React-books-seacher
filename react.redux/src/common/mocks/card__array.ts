export const mockCardArr = [
  {
    id: 'B096D6BCTF',
    title: 'The Unsinkable Greta James The Unsinkable Greta James The Unsinkable Greta James',
    authors: ['Jennifer E. Smith'],
    averageRating: 4.3,
    description:
      'Right after the sudden death of her mother—her first and most devoted fan—and just before the launch of her high-stakes sophomore album, Greta James falls apart on stage. The footage quickly goes viral and she stops playing, her career suddenly in jeopardy—the kind of jeopardy her father, Conrad, has always predicted; the kind he warned her about when he urged her to make more practical choices with her life.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B096HF9RJ4',
    title: 'All My Rage',
    authors: ['Sabaa Tahir'],
    averageRating: 4.7,
    description:
      'Misbah is a dreamer and storyteller, newly married to Toufiq in an arranged match. After their young life is shaken by tragedy, they come to the United States and open the Clouds Rest Inn Motel, hoping for a new start. ',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B092T8G139',
    title: 'Never Simple',
    authors: ['Liz Scheier'],
    averageRating: 4.3,
    description:
      'Scheier’s mother Judith was a news junkie, a hilarious storyteller, a fast-talking charmer you couldn’t look away from, a single mother whose devotion crossed the line into obsession, and—when in the grips of the mental illness that plagued every day of her life—a violent and abusive liar whose hold on reality was shaky at best. On an uneventful afternoon when Scheier was eighteen, her mother sauntered into the room to tell her two important things: one, she had been married for most of Scheier’s life to a man she’d never heard of, and two, the man she’d told Scheier was her father was entirely fictional.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B0927CQNWV',
    title: 'The Chase',
    authors: ['Candice Fox'],
    averageRating: 4.2,
    description:
      'With that, the largest manhunt in United States history is on. In response to a hostage situation, more than 600 inmates from the Pronghorn Correctional Facility, including everyone on Death Row, are released into the Nevada Desert. Criminals considered the worst of the worst, monsters with dark, violent pasts, are getting farther away by the second.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B096XGDWSX',
    title: 'In Love',
    authors: ['Amy Bloom'],
    averageRating: 4.6,
    description:
      'Amy Bloom began to notice changes in her husband, Brian: He retired early from a new job he loved; he withdrew from close friendships; he talked mostly about the past. Suddenly, it seemed there was a glass wall between them, and their long walks and talks stopped. Their world was altered forever when an MRI confirmed what they could no longer ignore: Brian had Alzheimer’s disease.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B097B2DXS8',
    title: 'Disorientation',
    authors: ['Elaine Hsieh Chou'],
    averageRating: 4,
    description:
      'Twenty-nine-year-old PhD student Ingrid Yang is desperate to finish her dissertation on the late canonical poet Xiao-Wen Chou and never read about “Chinese-y” things again. But after years of grueling research, all she has to show for her efforts are junk food addiction and stomach pain. When she accidentally stumbles upon a curious note in the Chou archives one afternoon, she convinces herself it’s her ticket out of academic hell.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
];
