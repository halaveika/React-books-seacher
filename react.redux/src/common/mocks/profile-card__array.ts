export const mockCardArr = [
  {
    id: 'test1-id',
    firstname: 'test1-firstname',
    lastname: 'test1-firstname',
    zipcode: '22555',
    birthday: '01/01/2001',
    gender: 'test1-gender',
    avatar: 'test1-avatar',
    city: 'test1-city',
    gifts: ['test1-gift1', 'test1-gift2'],
  },
  {
    id: 'test2-id',
    firstname: 'test2-firstname',
    lastname: 'test2-firstname',
    zipcode: '22555',
    birthday: '02/02/2002',
    gender: 'test2-gender',
    avatar: 'test2-avatar',
    city: 'test2-city',
    gifts: ['test2-gift1', 'test2-gift2'],
  },
  {
    id: 'test3-id',
    firstname: 'test3-firstname',
    lastname: 'test3-firstname',
    zipcode: '22555',
    birthday: '03/03/2003',
    gender: 'test3-gender',
    avatar: 'test3-avatar',
    city: 'test3-city',
    gifts: ['test3-gift1', 'test3-gift2'],
  },
];
