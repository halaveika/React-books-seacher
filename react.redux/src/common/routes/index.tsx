import React from 'react';
import AboutPage from '../../pages/about';
import MainPage from '../../pages/main';
import ProfilePage from '../../pages/profile';
import NotFoundPage from '../../pages/not-found';
import DetailsPage from '../../pages/details';

const routes = [
  { path: '/', element: <MainPage title="cards" /> },
  { path: '/about', element: <AboutPage title="about" /> },
  { path: '/profile', element: <ProfilePage title="order" /> },
  { path: '/details', element: <DetailsPage title="details" /> },
  {
    path: '/404',
    element: <NotFoundPage title="not found page" />,
  },
];

export default routes;
