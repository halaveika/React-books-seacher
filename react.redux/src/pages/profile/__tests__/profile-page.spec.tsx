import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ProfilePage from '../profile-page';
Enzyme.configure({ adapter: new Adapter() });
import { BrowserRouter } from 'react-router-dom';
import { mockCardArr } from '../../../common/mocks/profile-card__array';
import { FORM_CARDS } from '../../../common/constants';

describe('test ProfilePage component', () => {
  it('should render ProfilePage component', async () => {
    const wrapper = shallow<typeof ProfilePage>(
      <BrowserRouter>
        <ProfilePage title="test" />
      </BrowserRouter>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle callback in useEffect after umnount component', () => {
    const spyGetItem = jest
      .spyOn(Storage.prototype, 'getItem')
      .mockImplementation((key: string) => JSON.stringify(mockCardArr));
    const spySetItem = jest.spyOn(Storage.prototype, 'setItem');
    const wrapper = mount(
      <BrowserRouter>
        <ProfilePage title="test" />
      </BrowserRouter>
    );
    expect(spyGetItem).toBeCalled();
    wrapper.unmount();
    expect(spySetItem).toBeCalled();
  });
});
