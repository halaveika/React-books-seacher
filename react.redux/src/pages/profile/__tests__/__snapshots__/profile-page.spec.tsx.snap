// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProfilePage component should render ProfilePage component 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProfilePage
    title="test"
  />
</Router>
`;
